src/main/java/controller/ScreenController.java

```diff
@@ -91,15 +94,19 @@ public void updateUI() {
      * This function updates all locations of the sprites.
      * @param sprite Sprite that the location is updated from.
      */
-    private void update(SpriteBase sprite) {
-        ImageView image = images.get(sprites.indexOf(sprite));
-        image.relocate(sprite.getX(), sprite.getY());
-        if (sprite.getSpriteChanged()) {
-            image.setImage(new Image(
-            		getClass().getResource(sprite.getImagePath()).toExternalForm()));
-            sprite.setSpriteChanged(false);
-        }
-        image.setRotate(sprite.getR());
+    public void update(SpriteBase sprite) {
+    	int plaats = sprites.indexOf(sprite);
+    	if (plaats >= 0) {
```

Comment:
Why is this needed? If this is done because of errors, then the problem should be fixed where the update is called where it shouldn't be.

<bug